pkg/controller/import-controller.go

```diff
@@ -103,149 +185,128 @@ func (c *ImportController) Run(threadiness int, stopCh <-chan struct{}) error {
+
+	if len(podList) == 0 {
+		return nil, nil
+	} else if len(podList) > 1 {
+		return nil, errors.Errorf("multiple pods found for import PVC %s/%s", pvc.Namespace, pvc.Name)
+	}
+	return podList[0], nil
+}
+
 // Create the importer pod based the pvc. The endpoint and optional secret are available to
 // the importer pod as env vars. The pvc is checked (again) to ensure that we are not already
 // processing this pvc, which would result in multiple importer pods for the same pvc.
 func (c *ImportController) processPvcItem(pvc *v1.PersistentVolumeClaim) error {
-	ep, err := getEndpoint(pvc)
+
+	// find import Pod
+	pod, err := c.findImportPodFromCache(pvc)
 	if err != nil {
 		return err
 	}
-	secretName, err := getSecretName(c.clientset, pvc)
+	pvcKey, err := cache.MetaNamespaceKeyFunc(pvc)
 	if err != nil {
 		return err
 	}
-	if secretName == "" {
-		glog.V(Vadmin).Infof("no secret will be supplied to endpoint %q\n", ep)
-	}
 
-	// check our existing pvc one more time to ensure we should be working on it
-	// and to help mitigate any race conditions. This time we get the latest pvc.
-	doCreate, pvc, err := checkPVC(c.clientset, pvc, true)
-	if err != nil { // maybe an intermittent api error
-		return err
-	}
-	if !doCreate { // don't create importer pod but not an error
-		return nil // forget key; logging already done
+	// Pod must be controlled by this PVC
+	if pod != nil && !metav1.IsControlledBy(pod, pvc) {
+		return errors.Errorf("found pod %s/%s not owned by pvc %s/%s", pod.Namespace, pod.Name, pvc.Namespace, pvc.Name)
 	}
 
-	// all checks passed, let's create the importer pod!
-	pod, err := CreateImporterPod(c.clientset, c.importerImage, c.verbose, c.pullPolicy, ep, secretName, pvc)
-	if err != nil {
-		return err
+	// expectations prevent us from creating multiple pods. An expectation forces
+	// us to observe a pod's creation in the cache.
+	needsSync := c.podExpectations.SatisfiedExpectations(pvcKey)
+	if pod == nil && needsSync {
+		ep, err := getEndpoint(pvc)
+		if err != nil {
+			return err
+		}
+
+		secretName, err := getSecretName(c.clientset, pvc)
+		if err != nil {
+			return err
+		}
+		if secretName == "" {
+			glog.V(Vadmin).Infof("no secret will be supplied to endpoint %q\n", ep)
+		}
+
+		// all checks passed, let's create the importer pod!
+		c.expectPodCreate(pvcKey)
+		pod, err = CreateImporterPod(c.clientset, c.importerImage, c.verbose, c.pullPolicy, ep, secretName, pvc)
+		if err != nil {
+			c.observePodCreate(pvcKey)
+			return err
+		}
+		return nil
 	}
+
 	// update pvc with importer pod name and optional cdi label
-	anno := map[string]string{AnnImportPod: pod.Name}
+	anno := map[string]string{}
+	if pod != nil {
+		anno[AnnImportPod] = string(pod.Name)
+		anno[AnnPodPhase] = string(pod.Status.Phase)
+	}
+
 	var lab map[string]string
 	if !checkIfLabelExists(pvc, CDI_LABEL_KEY, CDI_LABEL_VALUE) {
 		lab = map[string]string{CDI_LABEL_KEY: CDI_LABEL_VALUE}
 	}
+
 	pvc, err = updatePVC(c.clientset, pvc, anno, lab)
```

Comment:
It seems like the call to `updatePVC` could happen when the pod is nil. If that's true the anno and label would be empty and not sure if that causes any problems.